src/home/components/components/components/fetch-pokemon-detail.ts

```diff
@@ -0,0 +1,18 @@
+import {useDispatch} from 'react-redux'
+import {savePokemonDetailAction} from '../pokemon-detail/redux/pokemonDetail.actions'
+
+export const  useFetchPokemonDetail = (url: string) => {
```

Comment:
Maybe the logic here can be improved because the Pokemon detail page will not load content after refresh.
Still you handled the case with a redirect to main page.